app/frontend/src/features/connections/friends/useFriendList.ts

```diff
@@ -0,0 +1,40 @@
+import { Error } from "grpc-web";
+import { useQuery, useQueries, UseQueryResult } from "react-query";
+import { User } from "../../../pb/api_pb";
+import { service } from "../../../service";
+
+function useFriendList() {
+  const { data: friendIds, error, isLoading, isError } = useQuery<
+    number[],
+    Error
+  >("friendIds", service.api.listFriends);
+
+  const friendQueries = useQueries<User.AsObject, Error>(
+    (friendIds ?? []).map((friendId) => {
+      return {
+        queryKey: ["user", friendId],
+        queryFn: () => service.user.getUser(friendId.toString()),
+        enabled: !!friendIds,
+      };
+    })
+  );
+
+  const hasErrors =
+    isError ||
+    !!(friendQueries.length && friendQueries.every((query) => query.isError));
+  const errors = [
+    error?.message,
+    ...friendQueries.map(
+      (query: UseQueryResult<unknown, Error>) => query.error?.message
+    ),
+  ].filter((e): e is string => typeof e === "string");
+
+  return {
+    isLoading: isLoading || friendQueries.some((query) => query.isLoading),
+    isError: hasErrors,
+    errors,
+    friendQueries,
```

Comment:
I think we should be returning the data rather than the queries. The info on the queries is covered by isLoading, isError and errors. Then it is more similar to `useQuery`

Reply:
Good point. Although that got me thinking, and I haven't done it at the moment, but do we want to ever display errors at a granular level? For example, some friends loaded successfully but 1-2 errored for some reason. 

Or do we only display their friends when everyone has successfully loaded, otherwise we display an error over the panel/tile?

Reply:
I think in this case it's not necessary.

Hmm... Unless the error is there's no connection. What happens in that case and there are still cached users? `isError` is true but data is defined? That is something I need to think about for `useUsers` too.

Reply:
If the returned data is a `Map<number, User.AsObject|undefoned>` we could display unloaded users as "(error loading user)" as well as the error message. 

Anyway do what you think is best/easiest for now. We will probably change this to use `useUsers` once that's merged and we can think about it then.

Reply:
Yeah `isError` will be true with data defined in case of cached user, which is a bit of a weird state to be in. I imagine in that case we'd want to keep displaying the user?

Only brought this up because of how I defined `isError` here too. I went with the optimistic definition of only saying there is an error if everything errored, but that definitely complicate things.

A blanket error might be the easiest approach for now (although might not be best UX?) and like you said we can re-visit later.